src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/debtposition/InstallmentProcessingService.java

```diff
@@ -0,0 +1,105 @@
+import it.gov.pagopa.pu.processexecutions.dto.generated.IngestionFlowFile;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+@Service
+@Lazy
+@Slf4j
+public class InstallmentProcessingService {
+
+    public static final String ORDINARY_SIL = "ORDINARY_SIL";
+    private final DebtPositionService debtPositionService;
+    private final InstallmentSynchronizeMapper installmentSynchronizeMapper;
+    private final InstallmentErrorsArchiverService installmentErrorsArchiverService;
+    private final DPInstallmentsWorkflowCompletionService dpInstallmentsWorkflowCompletionService;
+
+    public InstallmentProcessingService(DebtPositionService debtPositionService,
+                                        InstallmentSynchronizeMapper installmentSynchronizeMapper,
+                                        InstallmentErrorsArchiverService installmentErrorsArchiverService,
+                                        DPInstallmentsWorkflowCompletionService dpInstallmentsWorkflowCompletionService) {
+        this.debtPositionService = debtPositionService;
+        this.installmentSynchronizeMapper = installmentSynchronizeMapper;
+        this.installmentErrorsArchiverService = installmentErrorsArchiverService;
+        this.dpInstallmentsWorkflowCompletionService = dpInstallmentsWorkflowCompletionService;
+    }
+
+    /**
+     * Processes a stream of InstallmentIngestionFlowFileDTO and synchronizes each installment.
+     *
+     * @param iterator          Stream of installment ingestion flow file DTOs to be processed.
+     * @param ingestionFlowFile Metadata of the ingestion file containing details about the ingestion process.
+     * @param workingDirectory  The directory where error files will be written if processing fails.
+     * @return An {@link InstallmentIngestionFlowFileResult} containing details about the processed rows, errors, and archived files.
+     */
+    public InstallmentIngestionFlowFileResult processInstallments(Iterator<InstallmentIngestionFlowFileDTO> iterator,
+                                                                  IngestionFlowFile ingestionFlowFile,
+                                                                  Path workingDirectory) {
+        List<InstallmentErrorDTO> errorList = new ArrayList<>();
+        long processedRows = 0;
+        long totalRows = 0;
+
+        while (iterator.hasNext()) {
+            totalRows++;
+
+            InstallmentIngestionFlowFileDTO installment = iterator.next();
+            InstallmentSynchronizeDTO installmentSynchronizeDTO = installmentSynchronizeMapper.map(
+                    installment,
+                    ingestionFlowFile.getIngestionFlowFileId(),
+                    ingestionFlowFile.getOrganizationId()
+            );
```

Comment:
probably it cannot go into error, but let's move it inside the try/catch block ;)